### PR TITLE
New ram configuration

### PR DESCRIPTION
**Pull request by karsev**
@artokin

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -125,6 +125,9 @@ def buildStep(target, compilerLabel, toolchain, radioShield, meshInterface) {
             execute("sed -i '/mbed-mesh-api.thread-device-type\":/a \"mbed-mesh-api.thread-config-panid\": \"0xBAAB\",' ${config_file}")
             execute("sed -i 's/\"NANOSTACK\", \"THREAD_ROUTER\", \"COMMON_PAL\"/\"NANOSTACK\", \"THREAD_END_DEVICE\", \"COMMON_PAL\"/'  ${config_file}")
             execute("sed -i 's/\"mbed-mesh-api.thread-device-type\": \"MESH_DEVICE_TYPE_THREAD_ROUTER\"/\"mbed-mesh-api.thread-device-type\": \"MESH_DEVICE_TYPE_THREAD_MINIMAL_END_DEVICE\"/' ${config_file}")
+            if ("${toolchain}" == "GCC_ARM") {
+                execute("sed -i '/mbed-mesh-api.thread-device-type\":/a \"mbed-mesh-api.use-malloc-for-heap\": true,' ${config_file}")
```
Comment (SeppoTakalo) [inline Jenkinsfile]: I don't like this.
You are making changes into CI that are not happening for normal users.

Please modify the `mbed_app.json` files instead directly, and set this only for KW24D.

Our CI builds should be close match for normal user behaviour, like in OOB.

File: configs/mesh_thread.json [hunk 2]
```diff
@@ -32,6 +32,10 @@
         "NUCLEO_F401RE": {
             "LED": "LED_RED",
             "BUTTON": "USER_BUTTON"
+        },
+        "KW24D": {
+            "LED": "LED1",
+            "BUTTON": "SW1"
```
Comment (SeppoTakalo) [inline configs/mesh_thread.json]: Add here the `"mbed-mesh-api.use-malloc-for-heap": true`

**Comment by karsev** [comment]
Documentation updated. How we do with the jenkins configuration?

**Comment by SeppoTakalo** [comment]
LGTM, please use squashing when mergin in.

**Comment by MarceloSalazar** [comment]
Has this been tested with mbed OS 5.6?
I see this has been merged in master instead of the oob branch.

**Pull request merged**